### PR TITLE
Pulse 3257 - Enhanced Datadog Monitoring

### PR DESCRIPTION
**Pull request by rdempsey**
This PR adds enhanced Datadog monitoring to Traptor. 

**Testing (Trivialized. See Rob for details)**

- Create a VM
- Install Docker
- Install the Datadog agent using the test account.
- Add `non_local_traffic: yes` to the bottom of the `/etc/dd-agent/datadog.conf` file and restart the Datadog agent with `service datadog-agent restart`
- Check out this branch
- Fire up all things Dockerized Traptor (specifically Redis and Kafka)
- Add a rule to Redis and assign it to the running Traptor. Traptor will pick up on it and begin collection.
- Observe the metrics in Datadog
- Kill the Kafka container. Observe (after a few minutes) the error metrics in Datadog
- Bask in metric goodness

### CODE REVIEW

**Comment by jasonrhaas** [comment]
Same comment as https://github.com/istresearch/Cooper/pull/112

Lets continue to use `LogFactory` logging and adapter the DD logger to work with that.  The other problem is now I can't see this `dd_monitoring` when I'm testing locally since it relies on DataDog.  So I have different logging in development vs. production.

**Comment by rdempsey** [comment]
Whoomp there it is!

File: traptor/traptor.py [hunk 1]
```diff
@@ -153,6 +153,7 @@ def _setup_kafka(self):
                     'error_type': 'KafkaUnavailableError',
                     'ex': traceback.format_exc()
                 })
+                dd_monitoring.increment('traptor_error_occurred')
```
Comment (joequery) [inline traptor/traptor.py]: We should use an `error_type` datadog tag so we can aggregate on the types of errors that occur. I'm not sure if the increment function allows for tag in this rendition of dd_monitoring.py, but you could pull it over from Cooper.

File: traptor/traptor.py [hunk 2]
```diff
@@ -567,9 +574,21 @@ def _message_is_tweet(self, message):
...
+        Check if the message is a limit message.
+
+        :param message: message to check
+        :return: True if yes, False if no
+        """
+        if 'limit' in message:
+            return True
+        else:
+            return False
+
```
Comment (joequery) [inline traptor/traptor.py]: Can (optionally) be shortened to

```python
return 'limit' in message
```

File: traptor/dd_monitoring.py [hunk 3]
```diff
@@ -25,6 +25,10 @@
     'tweet_process_failure': 'traptor.src.tweet_process.failure',
     'tweet_to_kafka_success': 'traptor.src.tweet_to_kafka.success',
     'tweet_to_kafka_failure': 'traptor.src.tweet_to_kafka.failure',
+    'limit_message_received': 'traptor.src.limit.messages.count',
+    'limit_message_count': 'traptor.src.limit.current_limited',
+    'traptor_error_occurred': 'traptor.src.error.count',
+    'traptor_test': 'traptor.test'
```
Comment (rdempsey) [inline traptor/dd_monitoring.py]: Will do.

File: traptor/traptor.py (hunk 1 above)
Comment (rdempsey) [inline traptor/traptor.py]: Done.

File: traptor/traptor.py (hunk 2 above)
Comment (rdempsey) [inline traptor/traptor.py]: I likes to make it explicit.

**Pull request merged**